### DISCUSSION TITLE
Total consumption - alternative approach?

### DISCUSSION THREAD

**Discussion by O-Palda** [klacol/clage_homeserver#23] [Ideas]
Hello Klaus,

I have been experimenting with the API and maybe, if we sacrifice total number of watertaps, it would be possible to get the realtime totals of time, water adn energy in realtime while replacing the block "get consumption" 
https://github.com/klacol/clage_homeserver/blob/4e3feb40d142d967ada1254611dc00041fdac0ea/clage_homeserver/clage_homeserver.py#L317

with following:

url = "https://"+self.ipAddress+"/devices/logs/?showTotal=true"

which in my case returns immediately:

HTTP/1.1 200 OK
Access-Control-Allow-Origin: *
Connection: Keep-Alive
Content-Length: 411
Content-Type: application/json
Date: Sat, 27 Nov 2021 13:00:23 GMT

{
    "devices": [
        {
            "busId": 1,
            "connected": true,
            "id": "2049D933F3",
            "logs": [
                {
                    "cid": -1,
                    "id": -1,
                    "length": 300433,
                    "power": 939042,
                    "time": 1637014598,
                    "water": 3310495
                }
            ],
            "lqi": 0,
            "name": "Heater",
            "rev": 139,
            "rssi": 0,
            "signal": -65
        }
    ],
    "error": 0,
    "server": {
        "address": 100,
        "channel": 106,
        "id": "F5D002D261D5",
        "iot": {
            "state": "ONLINE"
        },
        "name": "DSX Server 63F5",
        "version": "1.0.17"
    },
    "success": true,
    "time": 1638018023,
    "version": "1.4"
}

while the original URL takes several seconds to respond with all 9000+ watertaps.

Kind regards,

Ondrej

**Comment by klacol** [comment]
Ok, good to know. The client-side counting (watertaps) and sum-up (energy) was made, because I could not find another solution. Great finding:

`"https://"+self.ipAddress+"/devices/logs/?showTotal=true"`

I will implement that soon.

**Comment by klacol** [comment]
Ich get the same result hier, with the call. So, we would replace the calls for the entity values as following:

`power `=> `consumption_energy` 
`water `=> `consumption_water `

I am not sure, what we have to map to `number_of_watertaps `and `usage_time`.

BTW: The name power in the API is wrong, since it is not related to power (kW) but energy (kWh). But we cannot change the API.

**Reply by O-Palda** [reply]
Hello Klaus,

well, I could live without number of watertaps and usagetime in home assistant.

Or, alternatively, you could use the integration counting for those two sensors with longer timeout and with lower frequency?
